### ISSUE TITLE
Color Model and API Questions

### ISSUE THREAD

**Issue by meodai** [words/color-description#1]
Hey @zeke & @dy 

For my color companion site I was experimenting with coming up with a color description for a color, based on its attributes in HSL for example.

I have a few questions:

### Description API: https://github.com/words/color-description/blob/main/index.js#L13

- This is a const right now and not configurable from the class itself. Is this something you see people extending easily? (If they do, I kinda want to know about it, and hope they would do a pull request). Is this wishful thinking or do you also see that as a "feature" ? :D

### Color Model: HSL

- HSL is not the best for an uniform hue perception, what space would you use to describe a colors hue range? HSLuv?
- I am a bit confused about how to handle that. On one and it is not an exact science anyway, its almost like a horoscope where the adjectives sometimes cover a wide range of colors, sometimes though the definition is quite narrow.

I would love to read about your thoughts.

**Comment by wooorm** [comment]
You might be interested in https://github.com/vasilisvg/human-colours

**Comment by zeke** [comment]
About seven years ago I created a project called https://github.com/colorjs/color-namer -- It vendors in a bunch of different datasets with names for colors. In hindsight, bringing all that data into the module feels like a mistake. I think it would have been better to have a uniform format for that and publish each set of names and color values as a standalone npm package.

>  what space would you use to describe a colors hue range?

I think the **Lab** or **HCL** color spaces may be what you're looking for: http://en.wikipedia.org/wiki/Lab_color_space

> Lab and HCL color spaces are special in that the perceived difference between two colors is proportional to their Euclidean distance in color space. This special property, called perceptual uniformity, makes them ideal for accurate visual encoding of data. In contrast, the more familiar RGB and HSL color spaces distort data when used for visualization.

**Comment by dy** [comment]
Regarding extensibility I can see i18n purpose for now.

**Comment by meodai** [comment]
@zeke ah I knew about your color-namer project. I run a similar project:  https://github.com/meodai/color-names. I'm happy you bring it up, because I think it is complementary. Those are more about a nice name for the color, what I aim to do with this project, is describing the attributes of the color using adjectives. And maybe give some insight about how the color is used in marketing or psychology...  (not sure about that thought)

About HCL: this is the model I had initially chosen, but it was hard to use because of cases like that:
<img width="956" alt="Screenshot 2021-05-12 at 12 34 03" src="https://user-images.githubusercontent.com/608386/117961456-6f24b700-b31e-11eb-8db6-49dfa54c5438.png">

It would make it hard to describe that range in a simple array like I did. But this made me think of HSLuv or HuSL! I will try something! Thanks for the insight.

@dy I think i18n would be a great fit! I will definitely integrate that!

@zeke & @dy I'd be a bit cautious when using the pantone color names in an open-source project. They are quite protective about it. A few github projects that were using the Pantone colors were taken down.

**Comment by dy** [comment]
Btw HSL has smoother chroma than HSLuv: 
![image](https://user-images.githubusercontent.com/300067/117968473-396ad880-b2f4-11eb-91af-324c68ac8bd1.png)
vs
![image](https://user-images.githubusercontent.com/300067/117968546-4c7da880-b2f4-11eb-9cb8-6440eea379f0.png)

That sharp blue saturation stands out in HSLuv:
![image](https://user-images.githubusercontent.com/300067/117968757-8b136300-b2f4-11eb-9cff-1b727ee4fdf7.png)
And HPLuv seems to be just a subrange of LCH.

**Comment by meodai** [comment]
@dy then HSL might not be such a bad choice after all. It doesn't matter if it's not a 100% accurate, because the adjectives are kinda vague anyway. (They are fitting for a range of color). I think its not tooo bad:

https://user-images.githubusercontent.com/608386/117972808-deed6e80-b32b-11eb-82c1-437a6d43169e.mov

I need to find a good way to test that.

**Comment by meodai** [comment]
@dy https://cdpn.io/meodai/debug/a7ab525f986069405732afd3d4f0e8a2

**Comment by dy** [comment]
Pretty cool! I wonder will that match canonical 2-component descriptors (color name + modifier), like _dark blue/gray/*_. That seems to be a [pattern for naming fwiw](https://en.wikipedia.org/wiki/List_of_colors:_A%E2%80%93F): _cosmic *_, _copper *_, _chinese *_, _cadmium *_, _burnt *_, _bright *_, _black *_ etc. So modifiers possibly indicate not only absolute range but also a relative shift from some canonic color name. Can be even vectorized and used to generate colors (more copper, more chinese, more cosmic etc.)

Also, in my head _warm_, _hot_, _red_, _reddish_ and _mellow_ are all different color categories.

**Comment by meodai** [comment]
@dy yeah that would be pretty cool and easy to pull off with that class and data structure.

**Comment by dy** [comment]
Possibly color temperature also may give an insight, tbh I imagined it more in this way https://eledlightslive.b-cdn.net/pub/media/magefan_blog/2016/04/color-temperature-chart.png

**Comment by meodai** [comment]
@dy I tried that first, but it made no sense for colors that are not blue yellow white orange or red :( // its styll in the code under temperature adjectives

**Comment by meodai** [comment]
the hue names should probably use LAB using this data:
https://uwdata.github.io/color-naming-in-different-languages/vis/full_color_maps.html and
https://uwdata.github.io/color-naming-in-different-languages/vis/stacked-spectrum.html

**Comment by meodai** [comment]
https://raw.githubusercontent.com/uwdata/color-naming-in-different-languages/master/model/colorSOM.json

@dy any idea what is meant by "color nodes" in the JSON data?

**Comment by dy** [comment]
It seems to be some data for the translator https://uwdata.github.io/color-naming-in-different-languages/vis/color_translator.html#term=pink&start_lang=en&dest_lang=en

**Comment by meodai** [comment]
@dy had the chance to continue to work on this a bit: https://words.github.io/color-description/

**Comment by zeke** [comment]
<img width="628" alt="Screen Shot 2021-09-11 at 4 08 49 PM" src="https://user-images.githubusercontent.com/2289/132963733-6d290b69-9970-4dc7-ac79-0312343c1ac1.png">

That's a lot of fun to play with!

**Comment by dy** [comment]
@meodai that is _so nice and fun_
![image](https://user-images.githubusercontent.com/300067/132969875-97a479bf-ad25-4784-94f9-a1808ce481b6.png)
As an idea: showing neighbor colors

**Comment by meodai** [comment]
@dy neighbors as in h+20, h-20 ?

**Comment by dy** [comment]
Yes, hue and maybe saturation.
I mean names are so fun but dense, picking is not precise, same time there are many almost-symonymic colors nearby with very different names- would be fun to know "color area", not only direct color.

**Comment by meodai** [comment]
@dy I like that idea! the "color area" thing would be hard to pull off. Feel free if you have an idea. But it would have to be visualized in 3d I think, since the "color area" is 3 dimensional.

**Comment by dy** [comment]
I'm thinking something simple:
![image](https://user-images.githubusercontent.com/300067/133106342-e3c7a08c-6b1d-4db0-b646-a5fdc5f3ff6a.png)

**Comment by meodai** [comment]
@dy cool idea! https://words.github.io/color-description/ I added it to the bottom thought, since this project is not really about the color names, but rather about the adjectives and descriptions ;)

**Comment by zeke** [comment]
☝🏼 That's really cool. I love clicking my way through the names and gradually changing from one hue to another.
